src/components/form.js

```diff
@@ -1,23 +1,71 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './books.css';
+import { useDispatch } from 'react-redux';
+import { nanoid } from '@reduxjs/toolkit';
+import { addBook } from '../redux/books/books';
 
-const Form = () => (
-  <>
-    <div className="border border-1 my-4" />
-    <section>
-      <h3>Add new book</h3>
-      <form className=" d-block d-md-flex gap-5 container-fluid m-0 p-0">
-        <input placeholder="Book title " className="Book-title" />
-        <input placeholder="Add Author " className="Book-title" />
-        <select className="dropdown">
-          <option value="action">action</option>
-          <option value="Science Fiction">Science Fiction</option>
-          <option value="Economy">Economy</option>
-        </select>
-        <button type="submit" className="primary-button-big border-0 rounded p-2 add-btn">ADD BOOK</button>
-      </form>
-    </section>
-  </>
-);
+const Form = () => {
+  // const [title, setTitle] = useState('');
+  // const [author, setAuthor] = useState('');
+
+  const dispatch = useDispatch();
+  const titleInput = useRef();
+  const authorInput = useRef();
+
+  const handleAddingBook = (e) => {
+    e.preventDefault();
+    const title = titleInput.current.value;
+    const author = authorInput.current.value;
+    if (title && author) {
+      const book = {
+        id: nanoid(),
+        title,
+        author,
```

Comment:
- [ ] [Required] Please note that the id `id: nanoid(),` is not getting saved to the redux state , when I add a new item it doesn't have an id which caused a small bug that whenever i delete a newly added book they all gonna be deleted (because they all don't have id) , kindly fix that 🙏 

image of the state:
![image](https://user-images.githubusercontent.com/89282221/216052877-89662c2a-bc3a-4048-a3d3-6d594d1388aa.png)

bug preview : 
![bug](https://user-images.githubusercontent.com/89282221/216053562-c98b387e-a6c5-4cdf-8676-30b37be0c404.gif)